data-structures/ds-stk/typescript/stack.ts

```diff
@@ -0,0 +1,53 @@
+type objected = undefined | object;
+
+class Stack{
+    private count : number = 0;
+    private storage: any = {};
+
+    public push = <T extends unknown>(item: T ): void  => {
+        this.storage[this.count] = item;
+        this.count++;
+
+    };
+
+    public pop = () : objected => {
+        if(this.count == 0){
+            return undefined;
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6362486)